README.md
# Scala Exercises - Circe library
------------------------

This repository hosts a content library for the [Scala Exercises](https://www.scala-exercises.org/) platform, including interactive exercises related to the [Circe](https://github.com/travisbrown/circe) JSON library.

## About Scala exercises

"Scala Exercises" brings exercises for the Stdlib, Cats, Shapeless and many other great libraries for Scala to your browser. Offering hundreds of solvable exercises organized into several categories covering the basics of the Scala language and it's most important libraries.

Scala Exercises is available at [scala-exercises.org](https://scala-exercises.org).

## Contributing

Contributions welcome! Please join our [Gitter channel](https://gitter.im/scala-exercises/scala-exercises)
to get involved, or visit our [GitHub site](https://github.com/scala-exercises).

## License

Copyright (C) 2015-2016 47 Degrees, LLC.
Reactive, scalable software solutions.
http://47deg.com
hello@47deg.com

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.


CHANGELOG.md
# Changelog

## [v0.6.6](https://github.com/scala-exercises/exercises-circe/tree/v0.6.6) (2020-08-10)

[Full Changelog](https://github.com/scala-exercises/exercises-circe/compare/v0.6.4...v0.6.6)

📈 **Dependency updates**

- Update definitions, exercise-compiler, ... to 0.6.4 [\#77](https://github.com/scala-exercises/exercises-circe/pull/77) ([47erbot](https://github.com/47erbot))
- Update sbt-mdoc to 2.2.4 [\#76](https://github.com/scala-exercises/exercises-circe/pull/76) ([47erbot](https://github.com/47erbot))
- Update scalacheck-1-14 to 3.2.1.0 [\#75](https://github.com/scala-exercises/exercises-circe/pull/75) ([47erbot](https://github.com/47erbot))
- Update scalatest to 3.2.1 [\#74](https://github.com/scala-exercises/exercises-circe/pull/74) ([47erbot](https://github.com/47erbot))
- Update sbt-scalafmt to 2.4.2 [\#73](https://github.com/scala-exercises/exercises-circe/pull/73) ([47erbot](https://github.com/47erbot))
- Update scalafmt-core to 2.6.4 [\#72](https://github.com/scala-exercises/exercises-circe/pull/72) ([47erbot](https://github.com/47erbot))
- Update scalafmt-core to 2.6.3 [\#71](https://github.com/scala-exercises/exercises-circe/pull/71) ([scala-steward](https://github.com/scala-steward))
- Update scalafmt-core to 2.6.2 [\#70](https://github.com/scala-exercises/exercises-circe/pull/70) ([scala-steward](https://github.com/scala-steward))

## [v0.6.4](https://github.com/scala-exercises/exercises-circe/tree/v0.6.4) (2020-06-30)

[Full Changelog](https://github.com/scala-exercises/exercises-circe/compare/v0.6.3...v0.6.4)

## [v0.6.3](https://github.com/scala-exercises/exercises-circe/tree/v0.6.3) (2020-06-29)

[Full Changelog](https://github.com/scala-exercises/exercises-circe/compare/v0.6.2...v0.6.3)

📈 **Dependency updates**

- Update definitions, exercise-compiler, ... to 0.6.3 [\#67](https://github.com/scala-exercises/exercises-circe/pull/67) ([scala-steward](https://github.com/scala-steward))
- Update sbt to 1.3.13 [\#66](https://github.com/scala-exercises/exercises-circe/pull/66) ([scala-steward](https://github.com/scala-steward))
- Update sbt-mdoc to 2.2.3 [\#65](https://github.com/scala-exercises/exercises-circe/pull/65) ([scala-steward](https://github.com/scala-steward))
- Update scalafmt-core to 2.6.1 [\#64](https://github.com/scala-exercises/exercises-circe/pull/64) ([scala-steward](https://github.com/scala-steward))
- Update scalacheck-1-14 to 3.2.0.0 [\#63](https://github.com/scala-exercises/exercises-circe/pull/63) ([scala-steward](https://github.com/scala-steward))
- Update scalatest to 3.2.0 [\#62](https://github.com/scala-exercises/exercises-circe/pull/62) ([scala-steward](https://github.com/scala-steward))
- Update sbt-mdoc to 2.2.2 [\#56](https://github.com/scala-exercises/exercises-circe/pull/56) ([scala-steward](https://github.com/scala-steward))
- Update sbt to 1.3.11 [\#49](https://github.com/scala-exercises/exercises-circe/pull/49) ([scala-steward](https://github.com/scala-steward))
- Update scalatest to 3.1.2 [\#47](https://github.com/scala-exercises/exercises-circe/pull/47) ([scala-steward](https://github.com/scala-steward))

**Merged pull requests:**

- Update scalafmt-core to 2.6.0 [\#60](https://github.com/scala-exercises/exercises-circe/pull/60) ([BenFradet](https://github.com/BenFradet))
- Update scalafmt-core to 2.5.3 [\#58](https://github.com/scala-exercises/exercises-circe/pull/58) ([BenFradet](https://github.com/BenFradet))
- Prepare repository for next  release and SBT build improvements [\#54](https://github.com/scala-exercises/exercises-circe/pull/54) ([juanpedromoreno](https://github.com/juanpedromoreno))
- tiny typo in the map definition [\#53](https://github.com/scala-exercises/exercises-circe/pull/53) ([dmarticus](https://github.com/dmarticus))
- change the typesignature for the mapArray method in transformJson [\#51](https://github.com/scala-exercises/exercises-circe/pull/51) ([dmarticus](https://github.com/dmarticus))

## [v0.6.2](https://github.com/scala-exercises/exercises-circe/tree/v0.6.2) (2020-04-27)

[Full Changelog](https://github.com/scala-exercises/exercises-circe/compare/v0.6.1...v0.6.2)

## [v0.6.1](https://github.com/scala-exercises/exercises-circe/tree/v0.6.1) (2020-04-27)

[Full Changelog](https://github.com/scala-exercises/exercises-circe/compare/v0.6.0...v0.6.1)

## [v0.6.0](https://github.com/scala-exercises/exercises-circe/tree/v0.6.0) (2020-04-24)

[Full Changelog](https://github.com/scala-exercises/exercises-circe/compare/b762749e79dae3f04484b54bc4a7ced7851c093b...v0.6.0)

**Merged pull requests:**

- Compile when Publish [\#40](https://github.com/scala-exercises/exercises-circe/pull/40) ([juanpedromoreno](https://github.com/juanpedromoreno))
- Updates Project [\#36](https://github.com/scala-exercises/exercises-circe/pull/36) ([juanpedromoreno](https://github.com/juanpedromoreno))
- Update sbt to 1.3.10 [\#34](https://github.com/scala-exercises/exercises-circe/pull/34) ([scala-steward](https://github.com/scala-steward))
- Update sbt to 1.3.9 [\#33](https://github.com/scala-exercises/exercises-circe/pull/33) ([scala-steward](https://github.com/scala-steward))
- Update scalacheck-shapeless\_1.14 to 1.2.5 [\#31](https://github.com/scala-exercises/exercises-circe/pull/31) ([scala-steward](https://github.com/scala-steward))
- Update scalatest to 3.1.1 [\#30](https://github.com/scala-exercises/exercises-circe/pull/30) ([scala-steward](https://github.com/scala-steward))
- Mergify: configuration update [\#29](https://github.com/scala-exercises/exercises-circe/pull/29) ([angoglez](https://github.com/angoglez))
- Update scalacheck-shapeless\_1.14 to 1.2.4 [\#28](https://github.com/scala-exercises/exercises-circe/pull/28) ([scala-steward](https://github.com/scala-steward))
- Update sbt to 1.3.8 [\#27](https://github.com/scala-exercises/exercises-circe/pull/27) ([scala-steward](https://github.com/scala-steward))

project/ProjectPlugin.scala
import sbt.Keys._
import sbt._
import com.alejandrohdezma.sbt.github.SbtGithubPlugin

object ProjectPlugin extends AutoPlugin {

  override def trigger: PluginTrigger = allRequirements

  override def requires: Plugins = plugins.JvmPlugin && SbtGithubPlugin

  object autoImport {

    lazy val V = new {
      val cats: String                = "2.1.1"
      val circe: String               = "0.13.0"
      val circeOptics: String         = "0.13.0"
      val circeGenericExtras: String  = "0.13.0"
      val shapeless: String           = "2.3.3"
      val scala: String               = "2.13.2"
      val scalaExercises: String      = "0.6.0"
      val scalacheck: String          = "1.14.3"
      val scalacheckShapeless: String = "1.2.5"
      val scalatest: String           = "3.1.2"
      val scalatestplusScheck: String = "3.1.1.1"
    }

    def dep(artifactId: String) = "org.scala-exercises" %% artifactId % V.scalaExercises

    lazy val exercisesSettings = Seq(
      libraryDependencies ++= Seq(
        dep("exercise-compiler"),
        dep("definitions"),
        "org.typelevel"              %% "cats-core"                 % V.cats,
        "io.circe"                   %% "circe-core"                % V.circe,
        "io.circe"                   %% "circe-generic"             % V.circe,
        "io.circe"                   %% "circe-parser"              % V.circe,
        "io.circe"                   %% "circe-generic-extras"      % V.circeGenericExtras,
        "io.circe"                   %% "circe-shapes"              % V.circe,
        "io.circe"                   %% "circe-optics"              % V.circeOptics,
        "com.chuusai"                %% "shapeless"                 % V.shapeless,
        "org.scalatest"              %% "scalatest"                 % V.scalatest,
        "org.scalacheck"             %% "scalacheck"                % V.scalacheck,
        "org.scalatestplus"          %% "scalacheck-1-14"           % V.scalatestplusScheck,
        "com.github.alexarchambault" %% "scalacheck-shapeless_1.14" % V.scalacheckShapeless
      ),
      addCompilerPlugin("org.typelevel" %% "kind-projector" % "0.11.0" cross CrossVersion.full)
    )
  }

  override def projectSettings: Seq[Def.Setting[_]] =
    Seq(
      organization := "org.scala-exercises",
      organizationName := "47 Degrees",
      organizationHomepage := Some(url("https://47deg.com")),
      scalaVersion := autoImport.V.scala,
      resolvers ++= Seq(
        Resolver.sonatypeRepo("snapshots"),
        Resolver.sonatypeRepo("releases")
      ),
      scalacOptions += "-Ymacro-annotations"
    )
}
